docs/src/man/basics.md

```diff
@@ -1510,32 +1510,96 @@ julia> german[Not(5), r"S"]
-
-The fundamental ways to specify a transformation are:
-
-- `source_column => transformation => target_column_name`;
-  In this scenario the `source_column` is passed as an argument to `transformation` function
-  and stored in `target_column_name` column.
-- `source_column => transformation`;
-  In this scenario we apply the transformation function to `source_column` and the
-  target column names is automatically generated.
-- `source_column => target_column_name` renames the `source_column` to `target_column_name`.
-- `source_column` just keep the source column as is in the result without any transformation;
-
-These rules are typically called transformation mini-language.
-
+In DataFrames.jl there are seven functions that can be used to transform data frame columns:
+
+| Function     | Memory Usage                     | Column Retention                             | Row Retention                                       |
+| ------------ | -------------------------------- | -------------------------------------------- | --------------------------------------------------- |
+| `transform`  | Creates a new data frame.        | Retains both source and transformed columns. | Retains same number of rows as source data frame.   |
+| `transform!` | Modifies an existing data frame. | Retains both source and transformed columns. | Retains same number of rows as source data frame.   |
+| `select`     | Creates a new data frame.        | Retains only transformed columns.            | Retains same number of rows as source data frame.   |
+| `select!`    | Modifies an existing data frame. | Retains only transformed columns.            | Retains same number of rows as source data frame.   |
+| `subset`     | Creates a new data frame.        | Retains both source and transformed columns. | Number of rows is determined by the transformation. |
+| `subset!`    | Modifies an existing data frame. | Retains both source and transformed columns. | Number of rows is determined by the transformation. |
+| `combine`    | Creates a new data frame.        | Retains only transformed columns.            | Number of rows is determined by the transformation. |
+
+### Constructing Transformation Pairs
+All of the functions above use the same syntax which is commonly
+`transformation_function(source_dataframe, transformation)`.
+The `transformation` argument is a `Pair` which defines the
+transformation to be applied to the `source_dataframe`,
+and it can take any of the forms listed below:
+
+- `source_column_selector => function`
+  - passes source column(s) to function
+  - automatically names the resulting column(s)
+- `source_column_selector => function => new_column_names`
+  - passes source column(s) to function
+  - names the resulting column(s) `new_column_names`
+    - *cannot be used with `subset` or `subset!`*
+- `source_column_selector => new_column_names`
+  - renames source column
+    - *may only select one column unless broadcasting*
```

Comment:
You mean `.=>` broadcasting? Then I wouldn't mention it as it's a different syntax from the one described in this point.

Reply:
Yes, okay.

Reply:
How about "*may only select one column (unless `.=>` broadcasting)*" or "*may only select one column*". @bkamins previously mentioned that I should add this point.